verifier.go

```diff
@@ -0,0 +1,99 @@
+package httpsignatures
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"fmt"
+	"strings"
+)
+
+// NewVerifier describes a function for creating a new instance of Verifier.
+type NewVerifier func(publicKey crypto.PublicKey, cryptoHash crypto.Hash) (Verifier, error)
+
+// newVerifier fetches the correct Verifier from the given signingAlgo. If an algo is provided,
+// it will be verified against the hashing algorithm in signingAlgo.
+func newVerifier(publicKey crypto.PublicKey, algo crypto.Hash, signingAlgo string) (Verifier, error) {
+	var signingHash string
+	var verifier NewVerifier
+	if strings.HasPrefix(signingAlgo, rsaName) {
```

Comment:
just a question: is it fine it's case sensitive?

Reply:
yes, I believe it is case sensitive